vscode/src/client/analyzerClient.ts

```diff
@@ -246,34 +249,33 @@ export class AnalyzerClient {
     }
 
     // Define the initialize request parameters
-    // TODO: With konveyor/kai#526, config.toml will be dropped.  The initialize parameters may
-    // TODO: change.  They'll need to be updated here.
-    const initializeParams = {
-      process_id: null,
-      kai_backend_url: getConfigKaiBackendURL(),
-      root_path: vscode.workspace.workspaceFolders![0].uri.fsPath,
-      log_level: getConfigLogLevel(),
-      log_dir_path: this.kaiDir,
-      model_provider: this.buildModelProviderConfig(),
-
-      file_log_level: getConfigLogLevel(),
-      demo_mode: this.isDemoMode(),
-      cache_dir: "",
-
-      // Analyzer and jdt.ls parameters
-      analyzer_lsp_rpc_path: this.getAnalyzerPath(),
-      analyzer_lsp_lsp_path: this.assetPaths.jdtlsBin,
-
-      // jdt.ls bundles (comma separated list of paths)
-      analyzer_lsp_java_bundle_path: this.assetPaths.jdtlsBundleJars.join(","),
-
-      // depOpenSourceLabelsFile
-      analyzer_lsp_dep_labels_path: this.assetPaths.openSourceLabelsFile,
+    const initializeParams: KaiRpcApplicationConfig = {
+      rootPath: vscode.workspace.workspaceFolders![0].uri.fsPath,
+      modelProvider: this.buildModelProviderConfig(),
+
+      logConfig: {
+        logLevel: getConfigLogLevel(),
+        fileLogLevel: getConfigLogLevel(),
+        logDirPath: this.kaiDir,
+      },
 
-      // TODO: Do we need to include `fernFlowerPath` to support the java decompiler?
-      // analyzer_lsp_fernflower: this.assetPaths.fernFlowerPath,
+      demoMode: this.isDemoMode(),
+
+      // Paths to the Analyzer and jdt.ls
+      analyzerLspRpcPath: this.getAnalyzerPath(),
+      analyzerLspLspPath: this.assetPaths.jdtlsBin,
+
+      // TODO: With konveyor/kai#509, multiple paths will be accepted
+      analyzerLspRulesPath: this.getRulesetsPath(),
+
+      // TODO: Once konveyor/kai#547 is resolved, multiple bundles can be supported
+      // jdt.ls bundles
+      analyzerLspJavaBundlePath: this.assetPaths.jdtlsBundleJars[0],
+
+      analyzerLspDepLabelsPath: this.assetPaths.openSourceLabelsFile,
 
-      analyzer_lsp_rules_path: this.getRulesetsPath(),
+      // TODO: Do we need to include `fernFlowerPath` to support the java decompiler?
+      // analyzerLspFernFlowerPath: this.assetPaths.fernFlowerPath,
```

Comment:
This makes me want to know two things.
1. Is the decompiler actually included in our assets? If it is we should just pass it on to kai. If not then, 👇 
1. @shawn-hurley or @pranavgaikwad are we doing any decompilation yet? My guess is no. If we don't have the asset and it isn't useful to us right now. Make an issue and we'll do it later.

Reply:
The provider support a cli argument of `fernFlowerPath` but that is not currently available in the analyzer rpc or the pass-through code.

The jar is currently in the repo here: https://github.com/konveyor/editor-extensions/blob/461e9ce7d7268137a2d2a45859b85c877019588d/assets/fernflower/fernflower.jar

If such a time happens where it is needed, it is easy to pass along.